### PR TITLE
Use OrderedSet to maintain DataFrame column order through set operations

### LINKED ISSUE #81: Use deterministic output order for column-based reporting
One section of the output report is called "Sample Rows with Unequal Values". It prints each column with differences and then 10 rows of differences. However, I've noticed that this output changes with multiple runs - so multiple calls of `.report()` will give you output in a different order. As an example, if you have two dataframes with the columns A, B, and C, all of which have differences, then one run may give you results in the order of B, C, A, the second in C, A, B, etc.

This isn't a huge issue - it's just cosmetic. But I do think the output would be cleaner if it the ordering was deterministic. Moreover, it should probably match the ordering of the columns of the first dataframe, which is referred to as `df1` in the code.

This section of the report is generated by looping through `self.column_stats` and generated sample rows of differences. `self.column_stats` is generated from the method `_intersect_compare()`, which loops through `self.intersect_columns()`. And finally, the issue is in `self.interesect_columns()`, which uses `set` - an unordered data structure - and does an intersection.

https://github.com/capitalone/datacompy/blob/8d96429c9cf574f8d448a4e63301f0025ad467ee/datacompy/core.py#L212-L214

I'm thinking the solution to this problem is use an ordered set. [There are packages available on PyPi for this](https://pypi.org/project/ordered-set/).

### PR DESCRIPTION
**Pull request by gandhis1**
@fdosani @KrishanBhasin 

Fixes #81. Iteration on #107. This aims to maintain the original order when printing out the set of common columns as well as unique columns, both which are based on set operations.

### CODE REVIEW

File: tests/test_core.py [hunk 1]
```diff
@@ -461,6 +461,27 @@ def test_columns_no_overlap():
...
+        [
+            (("A"), (0), (1), (2), (-1), (4), (-3)),
+            (("B"), (1), (2), (3), (-1), (4), (-2))
+        ],
+        columns=["join", "e", "h", "b", "a", "g", "d"]
+    )
+    compare = datacompy.Compare(df1, df2, ["join"])
+    assert list(compare.df1_unq_columns()) == ["f", "c"]
+    assert list(compare.df2_unq_columns()) == ["e", "d"]
+    assert list(compare.intersect_columns()) == ["join", "g", "b", "h", "a"]
```
Comment (gandhis1) [inline tests/test_core.py:482]: This example is intended to try to test edge cases (make sure the intersection is "stable" based on the left side), but doesn't necessarily reflect a realistic case. A realistic case is comparing two DataFrames where the columns are virtually identical with minimal differences, and it is in this context that maintaining the original DataFrame ordering actually becomes useful.

**Review by fdosani — COMMENTED** [review]
Thanks for the PR @gandhis1. @KrishanBhasin previous PR #107 partially fixes this correct. This is just a variation on the implementation from what I am seeing? In principal I'm fine with the changes, its more adding another dependency which I'd like to discuss. The library was last touched in June 2020. @jborchma @elzzhu thoughts?

**Comment by fdosani** [comment]
@gandhis1 Would you be open to implementing this without an external dependency? Maybe a `collections.OrderedDict` which just uses the keys. Talking about this with the team and I agree the order would be helpful but I'm a bit hesitant with introducing a new dependency like `ordered-set`

**Comment by gandhis1** [comment]
So we can copy in the source code here, which removes the PyPi package dependency. The library is under MIT license which as I understand means we just need a copyright and license notice on the file.

The challenge with using `OrderedDict` is it apparently does not support ordered set operations:

```
In [1]: x = OrderedDict([("b", None), ("a", None), ("c", None)])                                        

In [2]: y = OrderedDict([("c", None), ("b", None), ("z", None)])                                        

In [3]: x.keys()                                                                                        
Out[3]: odict_keys(['b', 'a', 'c'])

In [4]: x.keys() - y.keys()                                                                            
Out[4]: {'a'}

In [5]: type(x.keys() - y.keys())                                                                      
Out[5]: set

In [6]: x.keys() | y.keys()                                                                            
Out[6]: {'a', 'b', 'c', 'z'}
```

**Comment by gandhis1** [comment]
Revisiting this - any issue if I literally copy the original source code, which is a single file covered under the MIT license, into this repository? MIT license does not affect the license of any other code in this repository, as long as the original license is included with the file.

**Comment by fdosani** [comment]
@SanthiSridharan sorry for the extended delay on this. Been really busy.
So I think we can merge this in. the library doesn't introduce any sub-dependencies and I think that is fine in itself. If this causes issues we can revisit it at a later date.

**Review by fdosani — APPROVED** [review]
LGTM
